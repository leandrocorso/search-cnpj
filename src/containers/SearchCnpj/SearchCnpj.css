div.search-cnpj {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    height: 100%;
}

div.search-cnpj div.header {
    height: 5rem;
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 1rem;
    color: #eee;
    background-image: linear-gradient(to right, #967cf0, #a36cf3);
}

div.search-cnpj div.heading {
    display: flex;
    align-items: center;
}

div.search-cnpj .chart-icon {
    width: 1rem;
    height: 1rem;
    margin-right: 1rem;
    fill: #eee;
}

div.search-cnpj div.title {
    line-height: 120%
}

div.search-cnpj div.title small {
    display: block;
}

div.search-cnpj div.avatar {
    width: 2.5rem;
    height: 2.5rem;
    border-radius: 50%;
    background-image: url(https://yt3.ggpht.com/-o8QZ_nr7zlA/AAAAAAAAAAI/AAAAAAABfOs/cU3C7q5D8DA/s288-mo-c-c0xffffffff-rj-k-no/photo.jpg);
    background-size: cover;
}

div.search-cnpj div.step {
    display: flex;
    align-items: center;
    height: 4.5rem;
    padding: 1rem;
}

div.search-cnpj div.step div.step-number {
    display: flex;
    justify-content: center;
    width: 1rem;
    height: 1rem;
    margin-right: 1rem;
    font-size: 70%;
    font-weight: 700;
    color: #967cf0;
    border: 2px solid #967cf0;
}

/* Form */

div.search-cnpj form {
	display: flex;
	flex-direction: column;
	justify-content: space-between;
	height: 100%;
}

div.search-cnpj div.input-cnpj {
	display: block;
	margin-top: .5rem;
	font-size: 120%;
	font-weight: 700;
	color: #333;
}

div.search-cnpj div.input-cnpj:focus {
    color: #333;
}