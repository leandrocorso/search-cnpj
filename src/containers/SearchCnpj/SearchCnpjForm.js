import React, { Component } from 'react';
import styled from 'styled-components';

import { Container, Button, InputText } from '../../components';

import Services from '../../services';

const Form = styled.form`
	display: flex;
	flex-direction: column;
	justify-content: space-between;
	height: 100%;
`;

const InputCnpj = styled(InputText)`
	display: block;
	margin-top: .5rem;
	font-size: 120%;
	font-weight: 700;
	color: ${p => p.theme.themeDefault.colors.c8};

	&:focus {
		color: ${p => p.theme.themeDefault.colors.c8};
	}
`;

class SearchCnpjForm extends Component {

	constructor(props) {
		super(props);
		this.state = { isValid: '' }
	}

	getCnpj = async value => {
		this.setState({ isValid: false });
		const cnpj = value.replace(/\D/g, '');
		if (cnpj.length === 14) {
			try {
				const { isValid } = await Services.getCnpj(cnpj);
				this.setState({ isValid });
			} catch (e) {
				this.setState({ isValid: e.isValid });
			}
		}
	};

	render() {

		const { isValid } = this.state;

		return(
			<Form>
				<InputCnpj 
					id="cnpj" 
					name="cnpj" 
					label="CNPJ / Empresa" 
					placeholder="__.___.___/___-__"
					mask={[/\d/,/\d/,'.',/\d/,/\d/,/\d/,'.',/\d/,/\d/,/\d/,'/',/\d/,/\d/,/\d/,/\d/,'-',/\d/,/\d/,]}
					onChange={ this.getCnpj }
					autoComplete="off"
					isValid={isValid}
				/>
				<Container>
					<Button className="testButton" to="/" icon="arrow-right2" color={`${isValid ? 'success' : 'primary'}`}>Ok</Button>
				</Container>
			</Form>
		);
	}

}

export default SearchCnpjForm;
