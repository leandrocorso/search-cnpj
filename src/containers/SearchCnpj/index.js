import SearchCnpj from './SearchCnpj';

export default SearchCnpj;
