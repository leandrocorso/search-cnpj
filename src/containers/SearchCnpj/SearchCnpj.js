import React, { Fragment } from 'react';

import { Icon } from '../../components';

import SearchCnpjForm from './SearchCnpjForm';

import './SearchCnpj.css';

const SearchCnpj = () => (

    <div className="search-cnpj">

        <div className="header">

            <div className="heading">

                <Icon
                    className="chart-icon" 
                    data-test="chartIcon" 
                    src="line-chart" />

                <div data-test="title" className="title">
                    Nova Cotação <small data-test="subtitle">#980</small>
                </div>

            </div>

            <div data-test="avatar" className="avatar" />

        </div>

        <div className="step">

            <div 
                data-test="stepNumber"
                className="step-number">
                1
            </div>

            Busca por CNPJ ou empresa

        </div>

        <SearchCnpjForm/>

    </div>

);

export default SearchCnpj;
