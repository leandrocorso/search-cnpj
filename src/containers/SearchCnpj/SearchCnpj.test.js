import React from 'react';
import { shallow } from 'enzyme';

import SearchCnpj from './SearchCnpj';
import SearchCnpjForm from './SearchCnpjForm';

describe('Testing SearchCnpj component', () => {
    
    const wrapper = shallow(<SearchCnpj />);
    const wrapperForm = shallow(<SearchCnpjForm />);
    
    test('SearchCnpj exists', () => {
        expect(wrapper);
    });

    test('Should have "chart icon" in the heading', () => {
        expect(wrapper.find('.testHeadingIcon')).toHaveProp({ src : 'line-chart'});
    });

    test('Should number 1 into step indicator', () => {
        expect(wrapper.find('.testStep')).toHaveText('1');
    });

    test('Should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    // Form

    test('SearchCnpjForm exists', () => {
        expect(wrapperForm);
    });

    test('Input text should be named as "cnpj"', () => {
        expect(wrapperForm.find('#cnpj')).toHaveProp({ name: 'cnpj'});
    });

    test('Input text should be placeholded like cnpj format', () => {
        expect(wrapperForm.find('#cnpj')).toHaveProp({ placeholder : '__.___.___/___-__'});
    });

    test('Input text should be maked like cnpj format', () => {
        expect(wrapperForm.find('#cnpj')).toHaveProp({ mask : [/\d/,/\d/,'.',/\d/,/\d/,/\d/,'.',/\d/,/\d/,/\d/,'/',/\d/,/\d/,/\d/,/\d/,'-',/\d/,/\d/,] });
    });

    test('Input text should not enabled auto complete funcionality', () => {
        expect(wrapperForm.find('#cnpj')).toHaveProp({ autoComplete : 'off' });
    });

    test('Send button exists', () => {
        expect(wrapperForm.find('.testButton'));
    });

    test('Send button linked to home', () => {
        expect(wrapperForm.find('.testButton')).toHaveProp({ to : '/' });
    });

    test('Send button with right arrow icon', () => {
        expect(wrapperForm.find('.testButton')).toHaveProp({ icon : 'arrow-right2' });
    });

    test('Send button should be a color', () => {
        expect(wrapperForm.find('.testButton')).toHaveProp('color');
    });

    test('Should render correctly', () => {
        expect(wrapperForm).toMatchSnapshot();
    });

})