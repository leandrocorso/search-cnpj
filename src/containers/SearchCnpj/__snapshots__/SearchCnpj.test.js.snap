// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SearchCnpj component Should render correctly 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <Styled(Icon)
        className="testHeadingIcon"
        src="line-chart"
      />
      <styled.p>
        Nova Cotação 
        <small>
          #980
        </small>
      </styled.p>
    </styled.div>
    <styled.div />
  </styled.div>
  <styled.div>
    <styled.div
      className="testStep"
    >
      1
    </styled.div>
    Busca por CNPJ ou empresa
  </styled.div>
  <SearchCnpjForm />
</styled.div>
`;

exports[`Testing SearchCnpj component Should render correctly 2`] = `
<styled.form>
  <Styled(InputText)
    autoComplete="off"
    id="cnpj"
    isValid=""
    label="CNPJ / Empresa"
    mask={
      Array [
        /\\\\d/,
        /\\\\d/,
        ".",
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
        ".",
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
        "/",
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
        "-",
        /\\\\d/,
        /\\\\d/,
      ]
    }
    name="cnpj"
    onChange={[Function]}
    placeholder="__.___.___/___-__"
  />
  <styled.div>
    <Button
      className="testButton"
      color="primary"
      icon="arrow-right2"
      to="/"
    >
      Ok
    </Button>
  </styled.div>
</styled.form>
`;
